frontend/src/utilities/__tests__/clusterUtils.spec.ts

```diff
@@ -0,0 +1,87 @@
+import { useClusterInfo } from '~/redux/selectors/clusterInfo';
+import { getOpenShiftConsoleServerURL, useOpenShiftURL } from '~/utilities/clusterUtils';
+import { testHook } from '~/__tests__/unit/testUtils/hooks';
+const originalLocation = window.location;
+
+const setupWindowLocation = (hostname: string, protocol: string, port: string) => {
+  Object.defineProperty(window, 'location', {
+    writable: true,
+    value: { ...originalLocation, hostname, protocol, port },
+  });
+};
+
+jest.mock('~/utilities/const', () => ({
+  get DEV_MODE() {
+    return false;
+  },
+}));
+
+jest.mock('~/redux/selectors/clusterInfo', () => ({
+  ...jest.requireActual('~/redux/selectors/clusterInfo'),
+  useClusterInfo: jest.fn(),
+}));
+
+const useClusterInfoMock = useClusterInfo as jest.MockedFunction<typeof useClusterInfo>;
+const devModeMock = jest.spyOn(jest.requireMock('~/utilities/const'), 'DEV_MODE', 'get');
+
+describe('getOpenShiftConsoleServerURL', () => {
+  setupWindowLocation('localhost', 'https:', '443');
```

Comment:
This setup is only valid until the first test case changes the value.
Remove and make the call within the specific test case with values needed for that test case.